src/github.ts

```diff
@@ -199,47 +213,36 @@ export const release = async (
   const discussion_category_name = config.input_discussion_category_name;
   const generate_release_notes = config.input_generate_release_notes;
   try {
-    // you can't get a an existing draft by tag
-    // so we must find one in the list of all releases
-    if (config.input_draft) {
-      for await (const response of releaser.allReleases({
-        owner,
-        repo,
-      })) {
-        let release = response.data.find((release) => release.tag_name === tag);
-        if (release) {
-          return release;
```

Comment:
This was super duper confusing to me. It seemed like, if you came through to find a draft release (because `draft:true`) you just `return` out from here and skip all the logic below to update any of the fields. This means updating an existing draft release could only ever upload assets, it could not make any other changes. I don't know why you would want this and my PR changes this behavior. Now, if an existing release is found (no matter how) it is updated.